### PR TITLE
Use dataset

### PR DESCRIPTION
**Pull request by lannka**
To address comment from [here](https://github.com/ampproject/amphtml/pull/10438#discussion_r127516892)

### CODE REVIEW

File: src/3p-frame.js [hunk 1]
```diff
@@ -134,15 +134,14 @@ export function getIframe(
...
-  for (let i = 0; i < element.attributes.length; i++) {
-    const attr = element.attributes[i];
-    if (!startsWith(attr.name, 'data-')
-      // data-vars- is reserved for amp-analytics
-      // see https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/analytics-vars.md#variables-as-data-attribute
-      || startsWith(attr.name, 'data-vars-')) {
-      continue;
+  const dataset = element.dataset;
+  for (const name in dataset) {
+    if (dataset.hasOwnProperty(name)
```
Comment (jridgewell) [inline src/3p-frame.js]: Ooo, we should use this in the other places we use datasets.

**Comment by jridgewell** [comment]
Man, I just looked at MDN and it's IE11 only.

**Comment by lannka** [comment]
@jridgewell what is IE11 only? `dataset` is quite available: http://caniuse.com/#search=dataset

**Comment by jridgewell** [comment]
MDN [lists](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset#Browser_compatibility) only IE11. Caniuse says partial support since IE 6. 🤷‍♂️

File: src/3p-frame.js (hunk 1 above)
Comment (lannka) [inline src/3p-frame.js]: i actually think it's not necessary. i was mis-led by my IDE.

**Pull request merged**